Core/Src/main.c

```diff
@@ -87,8 +97,21 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
 
+  //chip select set to default high
+  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_SET);
+
+  //initialize PWM in channel 1
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);
+
+  //Same as {0b00000001, 0b10000000, 0b00000000}; from datasheet binary -> hex
+  const uint8_t transmit_data[3] = {0x01, 0x80, 0x00};
+  const uint8_t recieve_data[3] = {0};
```

Comment:
Close! 
1. If you want to make it const, try to follow the ALL_CAPS naming convention.
2. Technically here, `3` is a magic number, it would be best to make it a macro and give it a readable name (you can also use the macro for the size in the spi function instead of using the `sizeof` function)
3.  Are you sure you want to mark the recieve_data array as const? Its values will change.